common/util.go

```diff
@@ -441,18 +441,38 @@ func ReadAtMost(r io.Reader, n int) ([]byte, error) {
 }
 
 func detectNvidiaDevices() ([]string, error) {
+	nvidiaCardCount := 0
```

Comment:
nit: Why did you move this code up?

Reply:
Since `nvidiaCardCount` and `re` are used in both code branchings I preferred to take it out of the `GPU` path to highlight the shared usage. It may also be after `ghw.GPU()` so we allocate resources only if the later does not return an error.